crates/edr_napi/src/solidity_tests/op.rs

```diff
@@ -0,0 +1,67 @@
+use std::{collections::BTreeMap, sync::Arc};
+
+use edr_eth::Bytes;
+use edr_napi_core::solidity::{
+    config::{TestRunnerConfig, TracingConfigWithBuffers},
+    SyncTestRunner, SyncTestRunnerFactory,
+};
+use edr_op::solidity_tests::OpEvmBuilder;
+use edr_solidity::artifacts::ArtifactId;
+use edr_solidity_tests::{
+    contracts::ContractsByArtifact, decode::RevertDecoder, multi_runner::TestContract,
+    MultiContractRunner,
+};
+use napi::tokio;
+use napi_derive::napi;
+
+use crate::solidity_tests::{factory::SolidityTestRunnerFactory, runner::LazyContractDecoder};
+
+struct OpTestRunnerFactory;
+
+impl SyncTestRunnerFactory for OpTestRunnerFactory {
+    fn create_test_runner(
+        &self,
+        runtime: tokio::runtime::Handle,
+        config: TestRunnerConfig,
+        contracts: BTreeMap<ArtifactId, TestContract>,
+        known_contracts: ContractsByArtifact,
+        libs_to_deploy: Vec<Bytes>,
+        revert_decoder: RevertDecoder,
+        tracing_config: TracingConfigWithBuffers,
+    ) -> napi::Result<Box<dyn SyncTestRunner>> {
+        let contract_decoder = LazyContractDecoder::new(tracing_config);
+
+        let runner = runtime
+            .block_on(MultiContractRunner::<
```

Comment:
I'd wrap this in `block_in_place` to future proof it.

Reply:
Addressed by https://github.com/NomicFoundation/edr/pull/931/commits/f464009f6e47dd11e23bdacf2d593bcec9dab507